File: app/src/main/java/lishui/study/viewmodel/SquareViewModel.java
package lishui.study.viewmodel;

import android.app.Application;
import android.content.ContentUris;
import android.database.Cursor;
import android.graphics.Bitmap;
import android.net.Uri;
import android.provider.MediaStore;
import android.util.Size;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import lishui.study.common.log.LogUtil;
import lishui.study.common.util.BitmapUtils;
import lishui.study.common.util.Utilities;

public class SquareViewModel extends AndroidViewModel {

    private MutableLiveData<Bitmap> thumbnailLiveData = new MutableLiveData<>();

    public SquareViewModel(Application app) {
        super(app);
    }

    public void loadThumbnailBitmap() {
        try (Cursor cursor = MediaStore.Images.Media.query(
                getApplication().getContentResolver(),
                MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
                new String[]{MediaStore.Images.ImageColumns._ID,
                        MediaStore.Images.ImageColumns.ORIENTATION,
                        MediaStore.Images.ImageColumns.DATE_TAKEN},
                null,
                null,
                MediaStore.Images.ImageColumns.DATE_TAKEN + " DESC LIMIT 1")) {

            if (cursor != null && cursor.moveToNext()) {
                int idIndex = cursor.getColumnIndexOrThrow(MediaStore.Images.ImageColumns._ID);
                int orientationIndex = cursor.getColumnIndexOrThrow(MediaStore.Images.ImageColumns.ORIENTATION);

                int id = cursor.getInt(idIndex);
                int orientation = cursor.getInt(orientationIndex);

                LogUtil.d("get first thumbnail orientation: " + orientation);
                Bitmap thumb;
                if (Utilities.ATLEAST_Q) {
                    // get the thumb uri
                    final Uri thumbUri = ContentUris.withAppendedId(
                            MediaStore.Images.Media.EXTERNAL_CONTENT_URI, id);
                    // Thumbnails.MINI_KIND and signal can not be cancelled
                    thumb = getApplication().getContentResolver().loadThumbnail(thumbUri, new Size(512, 384), null);
                } else {
                    thumb = MediaStore.Images.Thumbnails.getThumbnail(getApplication().getContentResolver(),
                            id, MediaStore.Images.Thumbnails.MINI_KIND, null);
                }

                thumb = BitmapUtils.rotateBitmapIfNeed(thumb, 0, 0, thumb.getWidth(), thumb.getHeight(), orientation);

                thumbnailLiveData.postValue(thumb);
            }
        } catch (Exception e) {
            LogUtil.d("loadThumbnailBitmap error");
        }
    }

    public LiveData<Bitmap> getThumbnailLiveData() {
        return thumbnailLiveData;
    }
}

File: app/src/main/java/lishui/study/viewmodel/SquareViewModel.kt
package lishui.study.viewmodel

import android.app.Application
import android.content.ContentUris
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.provider.MediaStore
import android.util.Size
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import lishui.study.common.util.BitmapUtils
import lishui.study.common.util.Utilities

class SquareViewModel(val app: Application) : AndroidViewModel(app) {

    var thumbnailLiveData = MutableLiveData<Bitmap>()

    fun loadThumbnailBitmap() {
        viewModelScope.launch(Dispatchers.Default) {
            getFirstThumbnail()
        }
    }

    private fun getFirstThumbnail() {
        MediaStore.Images.Media.query(
                app.contentResolver,
                MediaStore.Images.Media.EXTERNAL_CONTENT_URI, arrayOf(MediaStore.Images.ImageColumns._ID,
                MediaStore.Images.ImageColumns.ORIENTATION,
                MediaStore.Images.ImageColumns.DATE_TAKEN),
                null,
                null,
                MediaStore.Images.ImageColumns.DATE_TAKEN + " DESC LIMIT 1").use { cursor ->
            if (cursor?.moveToFirst() == true) {
                val idIndex = cursor.getColumnIndexOrThrow(MediaStore.Images.ImageColumns._ID)
                val orientationIndex = cursor.getColumnIndexOrThrow(MediaStore.Images.ImageColumns.ORIENTATION)
                val id = cursor.getInt(idIndex)
                val orientation = cursor.getInt(orientationIndex)
                var thumb: Bitmap = MediaStore.Images.Thumbnails.getThumbnail(
                        app.contentResolver, id.toLong(), MediaStore.Images.Thumbnails.MINI_KIND, BitmapFactory.Options())
                 thumb = BitmapUtils.rotateBitmapIfNeed(thumb, 0, 0, thumb.width, thumb.height, orientation)
                thumbnailLiveData.postValue(null)
            }
        }
    }
}